### ISSUE TITLE
configure using Apache Reverse Proxy to run solely on port 80

### ISSUE THREAD

**Issue by marc-hanheide** [strands-project/strands_webtools#34]
Idea is to map different directories in Apache to the actual port requests, e.g.
- `<host:80>/rosweb` for the main page
- `<host:80>/rosweb/rosbridge` to act as a gateway to `localhost:9090` 
- `<host:80>/rosweb/mjpeg_stream` to act as a gateway to `localhost:8080` 
- [ ] This requires to modify the mjpegcanvas to work with URI that are not top-level.

**Comment by marc-hanheide** [comment]
It's not that easy :-)

The mjpeg_server streams using multipart/xmixed-replace MIME types which the Apache reverse proxy doesn't like out of the box. There are some posts suggesting that mod_rewrite could help here, but it doesn't look to promising. Also, for websockets to work, Apache needs a fix: http://stackoverflow.com/questions/14453123/apache-proxypassreverse-and-websockets

More to work on...

**Comment by marc-hanheide** [comment]
OK, I made some good progress on this. In principle I seem to be able to proxy all required connections now properly using mod_rewrite and mod_proxy when apache is patched (another module to tunnel ws connections) is being used as detailed here: http://blog.cafarelli.fr/post/2013/04/26/Backporting-Apache-support-for-websockets-reverse-proxy-(aka-getting-GateOne-to-work-behind-Apache) .
My test apache config looks like this:

```
<VirtualHost *:80>
DocumentRoot /var/www
ServerName marcs-pc
    RewriteEngine On
    LogLevel debug
    RewriteRule ^/video http://linda:8080/stream?topic=/head_xtion/rgb/image_color [P]
        ProxyPass /rosws ws://linda:9090/
        ProxyPassReverse /rosws ws://linda:9090/

</VirtualHost>
```

**Comment by marc-hanheide** [comment]
closed by #36
